documentation/examples/remote_storage/remote_storage_adapter/influxdb/client.go

```diff
@@ -77,6 +80,25 @@ func tagsFromMetric(m model.Metric) map[string]string {
 	return tags
 }
 
+// ensureDatabase ensures the existence of the database.
+func (c *Client) ensureDatabase() {
+	q := influx.Query{
+		Command: fmt.Sprintf(`CREATE DATABASE %s`, c.database),
```

Comment:
When authorization is turned on in InfluxDB, this query will require admin privileges, even if the database already exists. I tried running it with a user that has write permissions to the metrics database (which already exists), but isn't an admin. I got the following error:

```
ERR: error authorizing query: metrics not authorized to execute statement 'CREATE DATABASE metrics', requires admin privilege
```

I don't know if we should be mixing administrative database schema actions with normal user read/write access in this adapter. How about doing that externally instead? Otherwise we'll need to additionally pass in InfluxDB admin user credentials in here as well.

Reply:
> How about doing that externally instead?
what do you mean here?

Reply:
I mean ensuring the existence of a prerequisite databases in a separate process rather than this adapter. If we wanted to do it in here, we would have to make it optional and/or allow passing two different set of credentials (admin and normal read/write user). While that might be ok, I would point out that this is just example code, and we shouldn't make it too complex.

Reply:
exactly because it is an example , should we try to make it work for most people out of the box.

When I tried it I used the official influx docker image so didn't run into any permission issues. Probably  because it uses the admin user by default.

Reply:
without  this I had to exec in the container and create the DB which is few extra steps.
I am fine either way, so if you prefer to keep the example without this feel free to close it.